File: K47 User Manual/2_light_and_infrared_sensors/11_infraredEmitter/Code/Python/infrared.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import time

IrReceiverPin = 10
IrEmissionPin = 11
LedPin = 16

Led_status = 1

def setup():
	GPIO.setmode(GPIO.BOARD)       # Numbers GPIOs by physical location
	GPIO.setup(LedPin, GPIO.OUT)   # Set LedPin's mode is output
	GPIO.setup(IrReceiverPin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
	GPIO.setup(IrEmissionPin, GPIO.OUT)   # Set IrPin's mode is output
	GPIO.output(LedPin, GPIO.LOW) # Set LedPin low to off led

def irReceivedCallback(ev=None): # Called every time the irReceiver pin falls. Blink the LED!
	global Led_status
	Led_status = not Led_status
	GPIO.output(LedPin, Led_status)  # switch led status(on-->off; off-->on)
	print("received signal!\n")
	time.sleep(0.1)
	GPIO.output(LedPin, False)

def loop():
	GPIO.add_event_detect(IrReceiverPin, GPIO.FALLING, callback=irReceivedCallback) # wait for falling
	while True:
		print '...IrPin high'
		GPIO.output(IrEmissionPin, GPIO.HIGH)  # IrPin on
		time.sleep(0.5)
		print 'IrPin low...'
		GPIO.output(IrEmissionPin, GPIO.LOW) # IrPin off
		time.sleep(0.5)

def destroy():
	GPIO.output(LedPin, GPIO.LOW)     # led off
	GPIO.cleanup()                     # Release resource

if __name__ == '__main__':     # Program start from here
	setup()
	try:
		loop()
	except KeyboardInterrupt:  # When 'Ctrl+C' is pressed, the child program destroy() will be  executed.
                print("KeyboardInterrupt.\n")
		destroy()

File: K47 User Manual/2_light_and_infrared_sensors/11_infraredEmitter/Code/C/infrared.c
#include <wiringPi.h>
#include <stdio.h>

#define    IrEmissionPin    0
#define	   IrReceiverPin	16
#define	   LEDPin	4

int cnt = 0;

void infraredDetectionInterruptHandler(void)
// called at interrupt every time the IR receiver pin's signal falls to LOW

{
	printf("Received infrared signal. cnt = %d\n", ++cnt);
    
    // blink the LED briefly
	if(digitalRead(LEDPin) == HIGH)
	{
		digitalWrite(LEDPin, LOW);
	}
    else
	{
		digitalWrite(LEDPin, HIGH);
	}
    
	delay(100);
	digitalWrite(LEDPin, LOW);
}

int main(void)
{
	if(wiringPiSetup() == -1)
	{ 
		printf("setup wiringPi failed !\n");
		return -1; 
	}
	pinMode(LEDPin, OUTPUT);
	pinMode(IrEmissionPin, OUTPUT);
	pinMode(IrReceiverPin, INPUT);
	pullUpDnControl(IrReceiverPin, PUD_UP) 
	
    // register an interrupt service routine that will
    // be called every time the IR receiver pin goes LOW
    
	if(wiringPiISR(IrReceiverPin, INT_EDGE_FALLING, &infraredDetectionInterruptHandler) == -1)
	{
		printf("setup ISR failed !");
		return -1;
	}
    
    // now loop forever, from time to time blinking the IR
    // emitter ON and OFF.  If the receiver receives the emitter's emission,
    // then the interrupt routine should report to us "I saw it!" shortly after
    // this emitter logic tells us "I sent it!"
    
	while(1)
	{
		digitalWrite(IrEmissionPin, HIGH);
		printf("IrEmissionPin is set High\n");
		delay(500);
		
		digitalWrite(IrEmissionPin, LOW);
		printf("IrEmissionPin is set Low\n");
		delay(500);
	}
	
	return 0;
}
